File: java/src/main/java/com/hibou/sample/Insecure.java
package com.hibou.sample;

import java.io.IOException;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;
import java.util.Random;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

/**
 * INTENTIONALLY insecure patterns for scanner testing. Credentials are AWS's
 * public documentation EXAMPLE values (non-functional). Do not copy any of this
 * into production code. See ../SECURITY-FIXTURES.md.
 */
public final class Insecure {

    // Hardcoded credentials — secret scanners flag these.
    public static final String AWS_ACCESS_KEY_ID = "AKIAIOSFODNN7EXAMPLE";
    public static final String AWS_SECRET_ACCESS_KEY = "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY";

    // log4j 2.14.1 — Log4Shell (CVE-2021-44228). Logging attacker-controlled input
    // is the exact sink the CVE exploits.
    private static final Logger LOG = LogManager.getLogger(Insecure.class);

    private Insecure() {
    }

    /** Weak hashing (MD5). */
    public static String weakHash(String s) throws NoSuchAlgorithmException {
        MessageDigest md = MessageDigest.getInstance("MD5");
        byte[] digest = md.digest(s.getBytes());
        StringBuilder sb = new StringBuilder();
        for (byte b : digest) {
            sb.append(String.format("%02x", b));
        }
        return sb.toString();
    }

    /** Insecure, non-cryptographic RNG used as a "token". */
    public static int weakToken() {
        return new Random().nextInt();
    }

    /** Command injection — caller-controlled string handed to a shell. */
    public static Process runUserCommand(String userInput) throws IOException {
        return Runtime.getRuntime().exec(userInput);
    }

    /** Logs untrusted input through log4j — the Log4Shell sink. */
    public static void logUserInput(String userInput) {
        LOG.info("user said: {}", userInput);
    }
}

File: go/insecure.go
// insecure.go collects INTENTIONALLY insecure patterns so SAST scanners (gosec)
// and dependency scanners have findings to report. NONE of this is real: the
// credentials are AWS's public documentation EXAMPLE key (non-functional). Do not
// copy any of this into production code. See ../SECURITY-FIXTURES.md.
package calc

import (
	"crypto/md5"
	"fmt"
	"math/rand"
	"os/exec"

	jwt "github.com/dgrijalva/jwt-go"
	yaml "gopkg.in/yaml.v2"
)

// Hardcoded credentials (gosec G101). AWS-documented EXAMPLE values — non-functional.
const (
	awsAccessKeyID     = "AKIAIOSFODNN7EXAMPLE"
	awsSecretAccessKey = "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"
)

// WeakHash hashes s with MD5 (gosec G401/G501 — weak cryptographic primitive).
func WeakHash(s string) string {
	h := md5.Sum([]byte(s))
	return fmt.Sprintf("%x", h)
}

// WeakToken derives a token from math/rand (gosec G404 — weak, non-crypto RNG).
func WeakToken() int {
	return rand.Int()
}

// RunUserCommand executes an arbitrary shell command (gosec G204 — command injection).
func RunUserCommand(userInput string) ([]byte, error) {
	return exec.Command("sh", "-c", userInput).CombinedOutput()
}

// SignToken uses the unmaintained dgrijalva/jwt-go (CVE-2020-26160) — vulnerable dep.
func SignToken(subject string) (string, error) {
	tok := jwt.NewWithClaims(jwt.SigningMethodHS256, jwt.MapClaims{"sub": subject})
	return tok.SignedString([]byte(awsSecretAccessKey))
}

// ParseYAML uses gopkg.in/yaml.v2 v2.2.2 (CVE-2019-11253) — vulnerable dep.
func ParseYAML(in []byte) (map[string]interface{}, error) {
	out := map[string]interface{}{}
	err := yaml.Unmarshal(in, &out)
	return out, err
}

// AccessKeyID exposes the hardcoded key so it counts as "used" without weakening
// the gosec finding.
func AccessKeyID() string { return awsAccessKeyID }
